packages/voila/webpack.config.js

```diff
@@ -42,14 +60,28 @@ var distRoot = path.resolve(
 );
 
 module.exports = [
-  {
-    entry: ['./lib/index.js'],
+  merge(baseConfig, {
+    mode: 'development',
+    entry: ['./publicpath.js', './' + path.relative(__dirname, entryPoint)],
     output: {
-      filename: 'voila.js',
       path: distRoot,
-      libraryTarget: 'amd'
+      library: {
+        type: 'var',
+        name: ['_JUPYTERLAB', 'CORE_OUTPUT']
+      },
+      filename: 'bundle.js'
     },
-    module: { rules: rules },
-    devtool: 'source-map'
-  }
-];
+    plugins: [
+      new ModuleFederationPlugin({
+        library: {
+          type: 'var',
+          name: ['_JUPYTERLAB', 'CORE_LIBRARY_FEDERATION']
+        },
+        name: 'CORE_FEDERATION',
+        shared: {
+          ...data.dependencies
+        }
```

Comment:
This might be revisited at some point, based on the discussion and discussions in https://github.com/jupyterlab/jupyterlab/issues/9829